### PR TITLE
support linux

### PR DESCRIPTION
**Pull request by sowu880**
Support template management CLI in linux.

### CODE REVIEW

**Review by ACMoretxj — APPROVED** [review]

File: src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Utilities
+{
+    public static class OrasUtility
+    {
+        public static string GetOrasFileName()
+        {
+            string orasFileName;
+            switch (Environment.OSVersion.Platform)
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs:20]: Maybe we can use [IsOSPlatform()](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.isosplatform?view=netcore-3.1) here?

File: src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                default:
+                    throw new TemplateManagementException("Operation system is not supported");
+            }
+
+            return orasFileName;
+        }
+
+        public static async Task OrasExecutionAsync(string command, string orasFileName, string orasWorkingDirectory)
+        {
+            TaskCompletionSource<bool> eventHandled = new TaskCompletionSource<bool>();
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs:38]: Can we remove the GetOrasFileName function and just check the platform information during each execution? The caller should not care about whether windows or unix binary is used.

Comment (moria97) [reply src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs:38]: Seems we don't need this utility? We can put this logic in the OrasClient

File: src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            else
+            {
+                throw new OrasException(TemplateManagementErrorCode.OrasTimeOut, "Oras request timeout");
+            }
+        }
+
+        public static void AddFilePermissionInLinuxSystem(string fileName)
+        {
+            var command = $"chmod +x {fileName}";
+            var escapedArgs = command.Replace("\"", "\\\"");
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs:79]: In which case will we have quotes in the command?

File: src/Microsoft.Health.Fhir.TemplateManagement.UnitTests/OCIFileManagerTests.cs [hunk 4]
```diff
@@ -18,28 +18,44 @@ public class OCIFileManagerTests
...
-            string command = $"push {_testOneLayerImageReference} {_baseLayerTemplatePath} {_userLayerTemplatePath}";
-            OrasExecution(command);
+            string command = $"push {_testMultiLayersImageReference} {_baseLayerTemplatePath} {_userLayerTemplatePath}";
+            try
+            {
+                await OrasUtility.OrasExecutionAsync(command, _orasFileName, Directory.GetCurrentDirectory());
+            }
+            catch
+            {
+                _isOrasValid = false;
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.TemplateManagement.UnitTests/OCIFileManagerTests.cs:57]: Do we expect to skip or fail all tests when oras is invalid?

**Review by moria97 — CHANGES_REQUESTED** [review]

**Comment by sowu880** [comment]
> Shall we provide support for OSX (mac OS)? Seems a lot of people are using vs code on Mac

Yes, I think we could add a new task and realize in another PR, since we also need to setup OSX test environment in pipeline.

**Review by moria97 — APPROVED** [review]

**Pull request merged**